providers/source.rb

```diff
@@ -0,0 +1,54 @@
+#
+# Cookbok Name:: td-agent
+# Provider:: source
+#
+# Author:: Pavel Yudin <pyudin@parallels.com>
+#
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+use_inline_resources
+
+action :create do
+  fail 'You should set the node[:td_agent][:includes] attribute to true to use this resource.' unless node['td_agent']['includes']
+
+  service 'td-agent' do
+    action [ :nothing ]
+  end
+
+  template "/etc/td-agent/conf.d/#{new_resource.source_name}.conf" do
+    source 'source.conf.erb'
+    owner 'root'
+    group 'root'
+    mode '0644'
+    variables(type: new_resource.type,
+              params: new_resource.params,
+              tag: new_resource.tag)
+    cookbook 'td-agent'
+    notifies :reload, 'service[td-agent]'
```

Comment:
We cannot rely on `reload` since it is commented out in init script for Debian/Ubuntu :(
This should be fixed eventually, but cannot use it for now.

https://github.com/treasure-data/td-agent/blob/bbafcd59983546b9091933d51cd3239899dedb10/debian/td-agent.init#L156
https://github.com/treasure-data/omnibus-td-agent/blob/7dd6f47d1cf29d0b453252a9e7225db98abe5c76/templates/etc/init.d/deb/td-agent#L156